File: lib/make_prime_nums.py
#coding: utf-8

def eratosthenes(N):
    # 素数判定
    flag = True
    for i in range(2, int(N**0.5) + 1):
        if N % i == 0:
            flag = False
            break
    return flag

def make_prime_nums(N):
    # Nまでの素数のリストを作成
    prime_tf = [True for _ in range(N+1)]
    prime_tf[0] = False
    prime_tf[1] = False
    for i in range(N+1):
        if prime_tf[i]:
            if eratosthenes(i):
                for j in range(i*2, N+1, i):
                    prime_tf[j] = False
    primes = [i for i in range(N+1) if prime_tf[i]]
    return primes

File: nimlib/make_prime_nums.nim
import math, sequtils

proc eratosthenes(N:int):bool =
    result = true
    for i in 2..sqrt(N.toFloat).int:
        if N mod i == 0:
            result = false

proc make_prime_nums(N:int):seq =
    var
        prime_tf = newSeqWith(N+1, true)
        primes = newSeq[int]()
    prime_tf[0] = false
    prime_tf[1] = false
    for i in 0..N:
        if prime_tf[i]:
            if eratosthenes(i):
                for j in countup(i*2, N, i):
                    prime_tf[j] = false
    for i in 0..N:
        if prime_tf[i]:
            primes.add i
    return primes